### PR TITLE
Lakhveer/upgrading typescript

### PR DESCRIPTION
**Pull request by lakhveerk**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -15,6 +17,8 @@ module.exports = {
     '@typescript-eslint/no-namespace': 'off',
     '@typescript-eslint/explicit-member-accessibility': 'warn',
     '@typescript-eslint/interface-name-prefix': 'off',
+    '@typescript-eslint/ban-types': 'off',
```
Comment (erinha) [inline .eslintrc.js:20]: Do you have reasons for why we needed to turn off these two rules?

File: .eslintrc.js [hunk 2]
```diff
@@ -2,9 +2,11 @@ module.exports = {
   parser: '@typescript-eslint/parser', // Specifies the ESLint parser
   extends: [
     'plugin:@typescript-eslint/recommended', // Uses the recommended rules from the @typescript-eslint/eslint-plugin
-    'prettier/@typescript-eslint', // Uses eslint-config-prettier to disable ESLint rules from @typescript-eslint/eslint-plugin that would conflict with prettier
+    //'prettier/@typescript-eslint', // Uses eslint-config-prettier to disable ESLint rules from @typescript-eslint/eslint-plugin that would conflict with prettier
```
Comment (erinha) [inline .eslintrc.js:5]: Is there a reason why this is commented out? If we really don't need it anymore probably we should just delete this line?

Comment (lakhveerk) [reply .eslintrc.js:5]: All prettier configs have been merged into just 'prettier' I will delete the commented line

File: .eslintrc.js (hunk 1 above)
Comment (lakhveerk) [reply .eslintrc.js:20]: It gives ban-type errors in communication.ts files when we build the repo. We have these rules disabled in communication.ts in metaos repo. I will delete these lines from here, and will put in the communication file, so we wont be banning them for all files. Is that good?

**Review by ydogandjiev — APPROVED** [review]

**Review by sachingupta — APPROVED** [review]
please resolve merge conflict before completing this PR.

File: .eslintrc.js [hunk 3]
```diff
@@ -15,6 +16,8 @@ module.exports = {
     '@typescript-eslint/no-namespace': 'off',
     '@typescript-eslint/explicit-member-accessibility': 'warn',
     '@typescript-eslint/interface-name-prefix': 'off',
+    //'@typescript-eslint/ban-types': 'off',
```
Comment (erinha) [inline .eslintrc.js:19]: These changes are great! Think it's best to delete the commented out lines rather than comment them out please.

**Review by erinha — APPROVED** [review]

**Pull request merged**